nistats/first_level_model.py

```diff
@@ -583,6 +585,44 @@ def compute_contrast(self, contrast_def, stat_type=None,
 
         return outputs if output_type == 'all' else output
 
+    def get_voxelwise_model_attribute_(self, attribute, timeseries=True):
+
+        if self.minimize_memory:
+            raise ValueError('minimize_memory should be set to False to make residuals or predictions.')
+
+        if self.labels_ is None or self.results_ is None:
+            raise ValueError('The model has not been fit yet')
+
+        output = []
+
+        for design_matrix, labels, results in zip(self.design_matrices_, self.labels_, self.results_):        
+
+            if timeseries:
+                voxelwise_attribute = np.zeros((design_matrix.shape[0], len(labels)))
+            else:
+                voxelwise_attribute = np.zeros((1, len(labels)))
+
+            for label_ in results:
+                label_mask = labels == label_
+                voxelwise_attribute[:, label_mask] = getattr(results[label_], attribute)
+
+            output.append(self.masker_.inverse_transform(voxelwise_attribute))
+
+        if len(output) == 1:
+            return output[0]
+        else:
+            return output
+
+    def residuals(self):
+        return self.get_voxelwise_model_attribute_('resid')
+
+    def predicted(self):
+        return self.get_voxelwise_model_attribute_('predicted')
+
+    def rsq(self):
```

Comment:
Docstring.
is `rsq` already being used everywhere else in Nistats code? If not and it's a term you have introduced (or inherited from the PR you are working on) , a more explicit name for function and attribute will be better. For example `r_square` (f it is that) is a lot more self-documenting than `rsq`

Reply:
this was changed to r_square, resolving.